_rules/image-button-name-descriptive-f3b293.md

```diff
@@ -60,67 +57,68 @@ There is a known combination of a popular browser and assistive technology that
 
 ## Test Cases
 
 ### Passed
 
 #### Passed Example 1
 
-The image button has an [accessible name][] through the `alt` attribute.
+This `input` element with 'type' attribute in the `Image button` state has an [accessible name][] through the `alt` attribute that describes its purpose.
 
 ```html
 <input type="image" src="/test-assets/shared/search-icon.svg" alt="Search" />
 ```
 
 #### Passed Example 2
 
-The image button has an [accessible name][] through the `aria-label` attribute.
+The image button has an [accessible name][] through the `title` attribute that describes its purpose.
 
 ```html
-<input type="image" src="/test-assets/shared/search-icon.svg" aria-label="Search" />
+<input type="image" src="/test-assets/shared/search-icon.svg" title="Search" />
 ```
 
 #### Passed Example 3
 
-The image button has an [accessible name][] through the `title` attribute.
+The image button has an [accessible name][] through the `aria-label` attribute that describes its purpose.
 
 ```html
-<input type="image" src="/test-assets/shared/search-icon.svg" title="Search" />
+<input type="image" src="/test-assets/shared/local-search-icon.svg" aria-label="Search this page" />
 ```
 
 #### Passed Example 4
 
-The image button has an [accessible name][] through the `aria-labelledby` attribute.
+The image button has an [accessible name][] through the `aria-labelledby` attribute that describes its purpose.
 
 ```html
-<input type="image" src="/test-assets/shared/search-icon.svg" aria-labelledby="id1" />
-<div id="id1">Search</div>
+<h1 id="id1">Interesting buttons</h1>
+<input type="image" src="/test-assets/shared/share-icon.svg" name="Share" id="id2" aria-labelledby="id2 id1" />
 ```
 
 ### Failed
 
 #### Failed Example 1
 
-The image button element has an empty [accessible name][]. The `name` attribute can not be used to provide an [accessible name][].
+This `input` element with 'type' attribute in the `Image button` state has an empty [accessible name][], which cannot be descriptive. The `name` attribute does not provide an [accessible name][].
 
 ```html
 <input type="image" name="search" src="/test-assets/shared/search-icon.svg" />
 ```
 
 #### Failed Example 2
 
-The image button has an empty `alt` attribute, and no other attributes that can give it an [accessible name][].
+This image button has an `alt` attribute that is not descriptive as the [accessible name][].
 
 ```html
-<input type="image" src="/test-assets/shared/search-icon.svg" alt="" />
+<input type="image" src="/test-assets/shared/search-icon.svg" alt="Share" />
 ```
 
 #### Failed Example 3
 
-The image button has an `aria-labelledby` attribute, but the referenced element does not exist. This gives the button an empty [accessible name][].
+The image button has an `aria-labelledby` attribute, but the referenced element does not exist. This gives the button an empty [accessible name][], which cannot be descriptive.
```

Comment:
As with the other rule. These may actually have an accessible name according to the AAM.
OTOH, this is probably easier to get failed examples here given that the default name is "Submit Query"…

Reply:
I guess in this instance the default would not be descriptive.

Reply:
Yes, but the description should say so rather than saying that it has an empty accessible name.